lib/index_documents.rb

```diff
@@ -0,0 +1,86 @@
+require_relative "../app"
+require 'gds_api/publishing_api_v2'
+
+class IndexDocuments
+  def process(message)
+    raw_links = links_from_payload(message)
+
+    if !raw_links.nil? && !raw_links.empty?
+      links = remap_keys(raw_links, content_tagger_to_rummager_map)
+      links.each do |(link_type, content_ids)|
+        links[link_type] = paths_from_content_ids(content_ids)
+      end
+
+      base_path = document_base_path(message)
+      update_index(base_path, links)
+    end
+
+    message.ack
+  end
+
+private
+  def links_from_payload(message)
+    message.payload["links"].try(:slice, *content_tagger_tags)
+  end
+
+  def paths_from_content_ids(content_ids)
+    content_ids.map { |content_id| base_path content_id }.compact.sort
+  end
+
+  def content_tagger_tags
+    %w{
+      mainstream_browse_page
+      parent
+      topics
+      organisations
+    }
+  end
+
+  def content_tagger_to_rummager_map
+    {
+      "topics" => "specialist_sectors",
+      "organisations" => "lead_organisations",
```

Comment:
I don't think this is quite right; the field name in rummager is "organisations", too.  Elasticsearch does preserve the ordering of field values, though, so it might be worth ensuring that the value stored in the organisations field in elasticsearch has the "lead" organisations first, but are in alphabetical sorted order.  Something like (in pseudo code) "(organisations.sort + lead_organisations.sort).uniq".

[Aside: we might want to index the `lead_organisations` link type to a rummager field, but that would require adding a new field type in rummager, and besides nothing currently uses that, so that's beyond the scope of this PR.]